src/gui/sdlmain.cpp

```diff
@@ -846,28 +849,21 @@ Bitu GFX_SetSize(Bitu width,Bitu height,Bitu flags,double scalex,double scaley,G
 	}//CASE
 	if (retFlags)
 		GFX_Start();
-	if (!sdl.mouse.autoenable) SDL_ShowCursor(sdl.mouse.autolock?SDL_DISABLE:SDL_ENABLE);
 	return retFlags;
 }
 
-
-void GFX_CaptureMouse(void) {
-	sdl.mouse.locked=!sdl.mouse.locked;
-	if (sdl.mouse.locked) {
-		SDL_SetRelativeMouseMode(SDL_TRUE);
-		SDL_ShowCursor(SDL_DISABLE);
-	} else {
-		SDL_SetRelativeMouseMode(SDL_FALSE);
-		if (sdl.mouse.autoenable || !sdl.mouse.autolock) SDL_ShowCursor(SDL_ENABLE);
-	}
-        mouselocked=sdl.mouse.locked;
+SDL_bool mouse_is_captured = SDL_FALSE; // global for mapper
+void GFX_ToggleMouseCapture(void) {
+	SDL_ShowCursor(mouse_is_captured);
+	mouse_is_captured = mouse_is_captured ? SDL_FALSE : SDL_TRUE;
+	SDL_SetRelativeMouseMode(mouse_is_captured);
+	LOG_MSG("SDL: mouse is %s", mouse_is_captured ? "captured" : "released");
```

Comment:
It seems like `SDL_ShowCursor` changed from SDL 1.2 to SDL 2.0 in a subtle, (backwards compatible) way. This is good opportunity to use it according to [documentation](https://wiki.libsdl.org/SDL_ShowCursor); `SDL_ShowCursor` now accepts int and recognizes one of three values: SDL_QUERY, SDL_DISABLE, or SDL_ENABLE. HOWEVER - perhaps we don't need to invoke this function at all.

[`int SDL_SetRelativeMouseMode(SDL_bool)`](https://wiki.libsdl.org/SDL_SetRelativeMouseMode) can fail, so if it does, let's log the failure and error message and restore our internal value of `mouse_is_captured` field. I noticed in "Remarks" section:

> While the mouse is in relative mode, the cursor is hidden (…)

:) 

If it does indeed work, then I think it warrants a comment in code.

Reply:
Great find; and it works - both hiding and revealing the cursor on each toggle. 
Comment and error check added.